File: src/main/java/com/reelcosmos/entity/Actor.java
package com.reelcosmos.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "actors")
public class Actor extends BaseEntity {

    @Column(nullable = false, unique = true)
    private Long tmdbId;

    @Column(nullable = false)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String biography;

    private LocalDate birthday;

    private String placeOfBirth;

    private String profileImageUrl;

    @OneToMany(
            mappedBy = "actor",
            cascade = CascadeType.ALL,
            orphanRemoval = true
    )
    private List<MovieActor> movies = new ArrayList<>();
}

File: frontend/src/types/actor.ts
export interface Actor {

    id: number;

    tmdbId: number;

    name: string;

    biography: string;

    birthday: string;

    placeOfBirth: string;

    profileImageUrl: string;
}
export interface ActorPage {

    content: Actor[];

    totalPages: number;

    totalElements: number;

    size: number;

    number: number;

}
